validator_client/src/initialized_validators.rs

```diff
@@ -226,15 +189,15 @@ impl InitializedValidator {
                 }
 
                 // Append a `.lock` suffix to the voting keystore.
-                let voting_keystore_lockfile_path = get_lockfile_path(&voting_keystore_path)
+                let lockfile_path = get_lockfile_path(&voting_keystore_path)
                     .ok_or_else(|| Error::BadVotingKeystorePath(voting_keystore_path.clone()))?;
 
-                create_lock_file(&voting_keystore_lockfile_path, delete_lockfiles, &log)?;
+                let voting_keystore_lockfile = Lockfile::new(lockfile_path)?;
```

Comment:
Regarding backwards compat with existing versions of Lighthouse and Teku who also [use our existing strategy](https://github.com/sigp/lighthouse/issues/1823#issuecomment-716232791).

I understand with this PR we'd be unsafe (2 processes 1 file) if old-Lighthouse or Teku have created a lockfile. I guess this isn't a *huge* deal before genesis, but could possibly end up with a slashing.

One potential solution would be to add a `file_existed` field to `Lockfile` and log a `warn!` here if it is ``true`. It might be a handy little reminder to people that their node isn't exiting gracefully, as well.

The above solution doesn't ensure full safety, we'd have to basically revert the whole purpose of this PR achieve that (i.e., stop rebooting after a power failure).

Reply:
As discussed out of band we'll let this compatibility with Teku slip for now, as it's quite an obscure benefit (it only applies if the clients are sharing keystore directories, which they usually won't).

And "old-Lighthouse" nodes are still safe from slashing because they will have to use the same slashing protection DB, and there's an exclusive lock there too.